src/main/java/io/hops/util/FeaturestoreRestClient.java

```diff
@@ -488,5 +490,179 @@ public static FeaturestoreJdbcConnectorDTO getOnlineFeaturestoreJdbcConnectorRes
     JSONObject onlineFeaturestoreConnector = new JSONObject(responseEntity);
     return FeaturestoreHelper.parseJdbcConnectorJson(onlineFeaturestoreConnector);
   }
+  
+  /**
+   * Makes a REST call to Hopsworks to attach extended metadata to a
+   * featuregroup
+   *
+   * @param featuregroupName    the feature group name
+   * @param featurestore        the feature store
+   * @param featuregroupVersion the feature group version
+   * @param name                the extended attribute name
+   * @param value               the extended attribute value
+   * @throws FeaturestoreNotFound
+   * @throws JAXBException
+   * @throws FeaturegroupDoesNotExistError
+   * @throws FeaturegroupMetadataError
+   */
+  public static void addMetadata(String featuregroupName, String featurestore,
+      Integer featuregroupVersion, String name, String value)
+      throws FeaturestoreNotFound, JAXBException,
+      FeaturegroupDoesNotExistError, FeaturegroupMetadataError {
+    LOG.log(Level.FINE,
+        "Adding metadata to featuregroup: " + featuregroupName);
+    
+    JSONObject jsonObject = new JSONObject();
+    jsonObject.put(name, value);
+    
+    Response response;
+    try {
+      int featurestoreId = FeaturestoreHelper.getFeaturestoreId(featurestore);
+      int featuregroupId = FeaturestoreHelper.getFeaturegroupId(featurestore,
+          featuregroupName, featuregroupVersion);
+      response =
+          Hops.clientWrapper(jsonObject,
+              Constants.SLASH_DELIMITER +
+                  Constants.HOPSWORKS_REST_PROJECT_RESOURCE
+                  + Constants.SLASH_DELIMITER + Hops.getProjectId()
+                  + Constants.SLASH_DELIMITER
+                  + Constants.HOPSWORKS_REST_FEATURESTORES_RESOURCE +
+                  Constants.SLASH_DELIMITER +
+                  featurestoreId + Constants.SLASH_DELIMITER +
+                  Constants.HOPSWORKS_REST_FEATUREGROUPS_RESOURCE +
+                  Constants.SLASH_DELIMITER + featuregroupId +
+                  Constants.SLASH_DELIMITER +
+                  Constants.HOPSWORKS_FEATUREGROUPS_XATTRS_RESOURCE +
+                  Constants.SLASH_DELIMITER + name,
+              HttpMethod.PUT, null);
+    } catch (HTTPSClientInitializationException | JWTNotFoundException e) {
+      throw new FeaturestoreNotFound(e.getMessage());
+    }
+    LOG.log(Level.INFO,
+        "******* response.getStatusInfo():" + response.getStatusInfo());
```

Comment:
@maismail can we remove the `*` ?

Reply:
what do you mean? disallowing * as an xattr?

Reply:
no, just have some more meaningful logging instead of `********`

Reply:
Ahha, i merely copied the same structure from other methods in the class, to be consistent. https://github.com/logicalclocks/hops-util/blob/master/src/main/java/io/hops/util/FeaturestoreRestClient.java#L65.

Reply:
ok nvm